recipes/diligent-fx/all/conanfile.py

```diff
@@ -0,0 +1,117 @@
+
+    def package_id(self):
+        if self.settings.compiler == "Visual Studio":
+            if "MD" in self.settings.compiler.runtime:
+                self.info.settings.compiler.runtime = "MD/MDd"
+            else:
+                self.info.settings.compiler.runtime = "MT/MTd"
+
+    def validate(self):
+        if self.options.shared:
+            raise ConanInvalidConfiguration("Can't build as a shared lib")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+    def requirements(self):
+        self.requires("diligent-tools/cci.20211009")
+
+    def diligent_platform(self):
+        if self.settings.os == "Windows":
+            return "PLATFORM_WIN32"
+        elif self.settings.os == "Macos":
+            return "PLATFORM_MACOS"
+        elif self.settings.os == "Linux":
+            return "PLATFORM_LINUX"
+        elif self.settings.os == "Android":
+            return "PLATFORM_ANDROID"
+        elif self.settings.os == "iOS":
+            return "PLATFORM_IOS"
+        elif self.settings.os == "Emscripten":
+            return "PLATFORM_EMSCRIPTEN"
+        elif self.settings.os == "watchOS":
+            return "PLATFORM_TVOS"
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["DILIGENT_NO_FORMAT_VALIDATION"] = True
+        self._cmake.definitions["DILIGENT_BUILD_TESTS"] = False
+
+        self._cmake.definitions[self.diligent_platform()] = True
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def build(self):
+        self._patch_sources()
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        cmake = self._configure_cmake()
+        cmake.install()
+        self.copy("License.txt", dst="licenses", src=self._source_subfolder)
+        tools.rename(src=os.path.join(self.package_folder, "include", "source_subfolder"),
+                     dst=os.path.join(self.package_folder, "include", "DiligentFx"))
+        shutil.move(os.path.join(self.package_folder, "Shaders"), 
+                    os.path.join(self.package_folder, "res", "Shaders"))
+
+        self.copy(pattern="*.dll", src=self._build_subfolder, dst="bin", keep_path=False)
+        self.copy(pattern="*.dylib", src=self._build_subfolder, dst="lib", keep_path=False)
+        self.copy(pattern="*.lib", src=self._build_subfolder, dst="lib", keep_path=False)
+        self.copy(pattern="*.a", src=self._build_subfolder, dst="lib", keep_path=False)
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
```

Comment:
If possible, try to use the libraries names instead. Collect libs is fragile and does not respect linkage order.

Reply:
as I remember, there is only one lib for diligent-fx :)

Reply:
I have checked, there is only libDiligentFX.a.